src/openloco/windows/CompanyList.cpp

```diff
@@ -1,18 +1,1732 @@
+                    formatPerformanceIndex(company->performance_index, args);
+
+                    gfx::draw_string_494BBF(*dpi, 385, y - 1, 143, colour::black, stringId, &args);
+                }
+
+                {
+                    auto args = FormatArguments();
+
+                    args.push(string_ids::company_value_currency);
+                    args.push(company->companyValueHistory[0]);
+
+                    gfx::draw_string_494BBF(*dpi, 530, y - 1, 98, colour::black, stringId, &args);
+                }
+                y += 25;
+            }
+        }
+
+        // 0x00437AB6
+        static void sub_437AB6(window* self)
+        {
+            self->row_count = 0;
+
+            for (auto& company : companymgr::companies())
+            {
+                company.challenge_flags &= ~(1 << 3);
+            }
+        }
+
+        // 0x00436198
+        static void tabReset(window* self)
+        {
+            self->min_width = minWindowSize.width;
+            self->min_height = minWindowSize.height;
+            self->max_width = maxWindowSize.width;
+            self->max_height = maxWindowSize.height;
+            self->width = windowSize.width;
+            self->height = windowSize.height;
+            self->var_83C = 0;
+            self->row_hover = -1;
+            common::refreshCompanyList(self);
+        }
+
+        static void initEvents()
+        {
+            events.on_mouse_up = on_mouse_up;
+            events.on_resize = on_resize;
+            events.on_update = on_update;
+            events.event_08 = event_08;
+            events.event_09 = event_09;
+            events.get_scroll_size = get_scroll_size;
+            events.scroll_mouse_down = on_scroll_mouse_down;
+            events.scroll_mouse_over = on_scroll_mouse_over;
+            events.tooltip = tooltip;
+            events.cursor = cursor;
+            events.prepare_draw = prepare_draw;
+            events.draw = draw;
+            events.draw_scroll = draw_scroll;
+        }
+    }
+
+    // 0x00435BC8
+    window* open()
+    {
+        auto window = WindowManager::bringToFront(WindowType::companyList);
+
+        if (window != nullptr)
+        {
+            if (input::is_tool_active(_toolWindowType, _toolWindowNumber))
+            {
+                input::cancel_tool();
+                window = WindowManager::bringToFront(WindowType::messages);
+            }
+        }
+
+        if (window == nullptr)
+        {
+            gfx::ui_size_t windowSize = { 640, 272 };
+
+            window = WindowManager::createWindow(WindowType::companyList, windowSize, 0, &company_list::events);
+        }
+
+        //window->enabled_widgets = company_list::enabledWidgets;
+        //window->current_tab = 0;
+        window->frame_no = 0;
+        window->saved_view.clear();
+        window->flags |= window_flags::resizable;
+        window->var_83C = 0;
+        window->row_hover = -1;
+
+        company_list::sub_437AB6(window);
```

Comment:
You've changed the behaviour of the window just slightly. Before it wouldn't reset as many things when clicking the window open again.

Reply:
Sorry can you elaborate? I'm not sure I can see the difference.

Reply:
In the previous version the sort mode and a few other variables would not be reset when you reopen the window.

Reply:
Oh right, so the previous commit fixed that?

Reply:
No you still need to move some of these setting of values up into the new window branch of the code